packages/terminal/package.json

```diff
@@ -17,7 +17,10 @@
   },
   "theiaExtensions": [
     {
-      "frontend": "lib/browser/terminal-frontend-module",
+      "frontend": {
+        "path": "lib/browser/terminal-frontend-module",
+        "includeInSecondaryWindow": true
+      },
```

Comment:
I'm not really sure: the intention here is to load the CSS referenced from those modules in the secondary window. So it's kind of a second use of the same modules, whereas `frontend` and `frontendElectron` modules are different modules. Unless we want to introduce a new kind of module just for collecting CSS, we need to mark the existing modules for inclusion in `secondary-window.css` somehow.

Reply:
> So it's kind of a second use of the same modules, whereas frontend and frontendElectron modules are different modules.

As you can see in my suggested changes I am also referring to the same module in both places. From the code I've seen you are creating a new entry point for secondary windows and loading modules into it, so to me it makes sense to be able to refer to this new entry point as per my suggestion.

It keeps the declaration simple, even if we need to repeat the same module path for two entry points. Moreover it would allow declaring a module for secondary windows only, should that help.

Reply:
Also, do I understand correctly that we load Inversify module containers in that secondary window entry-point for the sole purpose of getting the CSS files required by them to load? And we don't care about any of the actual Inversify bindings?

Reply:
> we load Inversify module containers in that secondary window entry-point for the sole purpose of getting the CSS files required by them to load

That is correct

Reply:
> The trouble I see with repeating the same module path is that we really don't want a module to be loaded twice though, right (frontend, frontendElectron)?

I don't understand this concern. FYI the `frontendElectron` case is special: by default any module defined for `frontend` will also be included in Electron apps, except if `frontendElectron` is defined in which case it will use that in Electron instead.

In the current case, we don't seem to want to include `frontend` modules into secondary windows by default. So I believe we should get away with just adding another target which repeats the module path. It's fine because that module must be included in both entry points albeit only once per entry point, sure. I really prefer my suggestion as I believe it simplifies the shape of the JSON declaration and its processing somewhat.

FWIW I agree that it could make more sense to have something like you proposed:

```
"lib/browser/terminal-frontend-module": [ "frontend", "frontendSecondaryWindow" ]
```

But you're right that now might not be the time for that. Something to consider for later if we rework Theia extensions.